scripts/smooth-scrolling.js

```diff
@@ -8,33 +8,44 @@ $(function () {
         // Get link href
         var $anchor = $(this);
         var href = $anchor.attr('href');
-        var domain = new RegExp('/' + window.location.host + '/');
         if (
             href[0] !== '#' &&
-            domain.test(encodeURIComponent(href))
+            href.indexOf(window.location.host) == -1
         ) {
             return;
         }
 
         // This handles /path/current-page#element
-        href = href.split('#');
-        href = href.pop();
+        href = href.split('#').pop();
 
         // Get offset
         var scrollTop;
-        if (href === '#') {
+        if (href === '') {
             scrollTop = 0;
         } else {
-            scrollTop = $('#'+href).offset().top;
+            var $target = $('#'+href);
+            if (!$target.length) { // Anchor target not in this page
+                return;
+            }
+            scrollTop = $target.offset().top;
         }
 
         // Smooth scrolling
         $('html, body').stop().animate({
             scrollTop: scrollTop
         }, 'normal', function () {
-            window.location.hash = href;
+            window.location.hash = '#'+href;
         });
 
         event.preventDefault();
     });
+
+    // Fix anchors
+    // Let other scripts add links to DOM before fixing them
+    setTimeout(function () {
+        $('a[href^="#"]').each(function () {
+            var href = $(this).attr('href');
+            $(this).attr('href', window.location.pathname+href);
+        });
+    }, 0);
```

Comment:
>  `setTimeout(function () {}, 0)` re-queues the new JavaScript at the end of the execution queue.

See http://stackoverflow.com/questions/779379/why-is-settimeoutfn-0-sometimes-useful

It's the intended behavior since scripts executed after this one are adding other links to the DOM.